src/Compilers/Core/Portable/SourceGeneration/Nodes/SharedInputNodes.cs

```diff
@@ -14,6 +14,8 @@ internal static class SharedInputNodes
     {
         public static readonly InputNode<Compilation> Compilation = new InputNode<Compilation>(b => ImmutableArray.Create(b.Compilation));
 
+        public static readonly InputNode<CompilationOptions> CompilationOptions = new(b => ImmutableArray.Create(b.Compilation.Options));
```

Comment:
This is goodness, but we should double check with API review.

Reply:
note: this is in an internal type.  So this isn't explosed publicly yet.  We could certainly discuss that.  But it doesn't have to be part of this review :)